### PR TITLE
Eliminate more static uses of rewrite

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

File: src/theory/quantifiers/sygus/ce_guided_single_inv.cpp [hunk 1]
```diff
@@ -431,7 +431,12 @@ void CegSingleInv::setSolution()
   if (!d_solvedf.empty())
   {
     // replace the final solution into the solved functions
-    finalSol.applyToRange(d_solvedf, true);
+    finalSol.applyToRange(d_solvedf);
+    // rewrite the solution
+    for (size_t i = 0, nslv = d_solvedf.d_subs.size(); i < nslv; i++)
+    {
+      d_solvedf.d_subs[i] = rewrite(d_solvedf.d_subs[i]);
+    }
```
Comment (nafur) [inline src/theory/quantifiers/sygus/ce_guided_single_inv.cpp:439]:
```suggestion
    for (Node& n : d_solvedf.d_subs)
    {
      n = rewrite(n);
    }
```

**Review by nafur — APPROVED** [review]
LGTM.

File: src/theory/quantifiers/sygus/ce_guided_single_inv.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/quantifiers/sygus/ce_guided_single_inv.cpp:439]: I don't think this updates the vector, right?

Comment (nafur) [reply src/theory/quantifiers/sygus/ce_guided_single_inv.cpp:439]: It does. The Node is taken as a non-const reference and is updated in the loop body.

Comment (ajreynol) [reply src/theory/quantifiers/sygus/ce_guided_single_inv.cpp]: I see, yeah, its implicitly a double pointer.

**Pull request merged**